### PR TITLE
Add integration tests "Tester" component.

### PR DESCRIPTION
**Pull request by Creatone**
Below command to run Tester:
`python owca/main.py -c configs/extra/tester_example.yaml -r owca.extra.tester:Tester -r owca.extra.tester:MetricCheck -r owca.extra.tester:FileCheck --log=debug --root`
(Can be run on Ubuntu as well.)

### CODE REVIEW

File: owca/extra/tester.py [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+
+    with open('{}/tasks'.format(cpu_path), 'a') as f:
+        f.write(str(p.pid))
+    with open('{}/tasks'.format(perf_path), 'a') as f:
+        f.write(str(p.pid))
+
+    return p
+
+
+def _get_cgroup_full_path(cgroup):
```
Comment (Creatone) [inline owca/extra/tester.py]: use os.path.join

File: owca/extra/tester.py [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+
+    return name, name, '/{}'.format(task)
+
+
+def _create_dumb_process(cgroup_path):
+    command = ['sleep', 'inf']
+    p = subprocess.Popen(command)
+    cpu_path, perf_path = _get_cgroup_full_path(cgroup_path)
+
+    with open('{}/tasks'.format(cpu_path), 'a') as f:
```
Comment (Creatone) [inline owca/extra/tester.py]: use os.path.join

File: owca/extra/tester.py [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+        f.write(str(p.pid))
+    with open('{}/tasks'.format(perf_path), 'a') as f:
```
Comment (Creatone) [inline owca/extra/tester.py]: use os.path.join

File: owca/extra/tester.py [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+        self.test_current = 0
+        self.test_number = len(self.config_data)
+        self.metrics = []
+        self.processes: List[subprocess.Popen] = []
+        self.tasks = []
+
+    def get_tasks(self) -> List[Task]:
+
+        self.test_current += 1
+
```
Comment (Creatone) [inline owca/extra/tester.py]: do checks here

File: owca/extra/tester.py [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+            for check_case in test_case['checks']:
+                check_case: Check
+                try:
+                    check_case.check()
+                except CheckFailed as fail:
+                    # Clean tests processes and cgroups after failure.
+                    self._clean()
+                    raise CheckFailed(fail.args[0])
+
+            self._clean()
```
Comment (Creatone) [inline owca/extra/tester.py]: keep needed tasks, do not clean all

File: owca/extra/tester.py [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+
+        self.tasks = []
+
+        for task_name in test_case['tasks']:
```
Comment (Creatone) [inline owca/extra/tester.py]: create only new one

File: owca/extra/static_allocator.py [hunk 7]
```diff
@@ -145,28 +142,30 @@ def allocate(
             tasks_labels: TasksLabels,
             tasks_allocations: TasksAllocations,
     ) -> (TasksAllocations, List[Anomaly], List[Metric]):
-        if not os.path.exists(self.config):
-            log.warning('StaticAllocator: cannot find config file %r - ignoring!', self.config)
+
```
Comment (felidadae) [inline owca/extra/static_allocator.py]: I dont see point in settings rules in constructor when also config is provided.
I think it should be or rules or config:
assert(rules is None and config is not None) or (rules is not None and config is None)
@ppalucki  what You think?
Also please update docstring, once it is decided.

Comment (felidadae) [reply owca/extra/static_allocator.py]: discussed online, rejected

File: owca/extra/static_allocator.py [hunk 8]
```diff
@@ -128,14 +128,11 @@ class StaticAllocator(Allocator):
     If there are multiple matching rules then the rules' allocations are merged and applied.
     """
 
-    # File location of yaml config file with rules.
-    config: str
+    # Rules.
```
Comment (Damenus) [inline owca/extra/static_allocator.py]: This comment doesn't give any information. I think it can be removed.

File: owca/extra/tester.py [hunk 9]
```diff
@@ -0,0 +1,235 @@
...
+    def __post_init__(self):
+        self.config_data = load_config(self.config)['tests']
+        self.test_current = 0
+        self.test_number = len(self.config_data)
+        self.metrics = []
+        self.processes: Dict[str, subprocess.Popen] = {}
+        self.tasks = []
+        self.checks = []
+
+    def get_tasks(self) -> List[Task]:
```
Comment (ppalucki) [inline owca/extra/tester.py]: add unittest only for this method without need and call it at least 3 times 
make sure that internal state is correct between calls is correct and after third call sys.exit mocked was called

you can mock Checks as well and make sure there were called

File: Pipfile.lock [hunk 10]
```diff
@@ -211,11 +211,11 @@
         },
```
Comment (ppalucki) [inline Pipfile.lock]: drop this change restarto old pipefile lock unless you reall need it

**Comment by felidadae** [comment]
merge with master, as jenkins cannot test it

File: owca/extra/tester.py [hunk 11]
```diff
@@ -0,0 +1,238 @@
...
+
+    def __post_init__(self):
+        self.config_data = load_config(self.config)['tests']
+        import IPython; IPython.embed()
+        self.test_current = 0
+        self.test_number = len(self.config_data)
+        self.metrics = []
+        self.processes: Dict[str, subprocess.Popen] = {}
+        self.tasks = []
+        self.checks = []
```
[the remark this answers is not in the document]
Comment (Creatone) [reply owca/extra/tester.py]: Yes

File: owca/extra/tester.py [hunk 12]
```diff
@@ -0,0 +1,238 @@
...
+
+        for task_name in tasks_to_check:
+            name, task_id, cgroup_path = _parse_task_name(task_name)
+            subgroups_paths = []
+            labels = {}
+            resources = {}
+            task = Task(name, task_id, cgroup_path, subgroups_paths, labels, resources)
+            _create_cgroup(cgroup_path)
+
+            if self.command:
```
[the remark this answers is not in the document]
Comment (Creatone) [reply owca/extra/tester.py]: Command is passed via yaml

**Pull request merged**